lib/plugins/paragon-webpack-plugin/ParagonWebpackPlugin.js

```diff
@@ -0,0 +1,156 @@
+const { Compilation, sources } = require('webpack');
+const dotenv = require('dotenv');
+const fs = require('fs');
+const path = require('path');
+const {
+  getParagonVersion,
+  getParagonThemeCss,
+} = require('../../../config/data/paragonUtils');
+const {
+  injectMetadataIntoDocument,
+  // getMfeRuntimeConfig,
+  getParagonStylesheetUrls,
+  injectParagonCoreStylesheets,
+  injectParagonThemeVariantStylesheets,
+} = require('./utils');
+const resolvePrivateEnvConfig = require('../../resolvePrivateEnvConfig');
+
+// Add process env vars. Currently used only for setting the
+// server port and the publicPath
+dotenv.config({
+  path: path.resolve(process.cwd(), '.env.development'),
+});
+
+// Allow private/local overrides of env vars from .env.development for config settings
+// that you'd like to persist locally during development, without the risk of checking
+// in temporary modifications to .env.development.
+resolvePrivateEnvConfig('.env.private');
+
+// Resolve configuration `env.config`.
+const envConfigFile = path.resolve(process.cwd(), 'env.config.js');
+
+// Get Paragon and brand versions / CSS files from disk.
+const paragonVersion = getParagonVersion(process.cwd());
+const paragonThemeCss = getParagonThemeCss(process.cwd());
+const brandVersion = getParagonVersion(process.cwd(), { isBrandOverride: true });
+const brandThemeCss = getParagonThemeCss(process.cwd(), { isBrandOverride: true });
+
+/**
+ * 1. Injects `PARAGON_THEME` global variable into the HTML document during the Webpack compilation process.
+ * 2. Injects `<link rel="preload" as="style">` element(s) for the Paragon and brand CSS into the HTML document.
+ */
+class ParagonWebpackPlugin {
+  constructor({ processAssetsHandlers = [] } = {}) {
+    this.pluginName = 'ParagonWebpackPlugin';
+    this.paragonThemeUrlsConfig = {};
+    this.paragonMetadata = {};
+
+    // List of handlers to be executed after processing assets during the Webpack compilation.
+    this.processAssetsHandlers = [
+      this.resolveParagonThemeUrlsFromConfig,
+      this.injectParagonMetadataIntoDocument,
+      this.injectParagonStylesheetsIntoDocument,
+      ...processAssetsHandlers,
+    ].map(handler => handler.bind(this));
+  }
+
+  /**
+   * Resolves the MFE configuration for ``PARAGON_THEME_URLS`` in the following priority order:
+   *
+   * 1. ``mfeRuntimeApiConfig.PARAGON_THEME_URLS``
+   * 2. ``envConfig.PARAGON_THEME_URLS``
+   * 3. ``process.env.PARAGON_THEME_URLS``
+   *
+   * @returns {Object} Metadata about the Paragon and brand theme URLs from configuration.
+   */
+  async resolveParagonThemeUrlsFromConfig() {
+    // const mfeRuntimeApiConfig = await getMfeRuntimeConfig();
+    let envConfig;
+    if (fs.existsSync(envConfigFile)) {
+      envConfig = require(envConfigFile);
+    }
+    // const paragonThemeUrls = mfeRuntimeApiConfig?.PARAGON_THEME_URLS
```

Comment:
Hello, I would suggest removing these comments if they are not intended to be used in the future.